rpc/identity_provider.go

```diff
@@ -54,3 +67,91 @@ func (s *RPC) handleJWKS(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	json.NewEncoder(w).Encode(pkd)
 }
+
+func (s *RPC) getIDToken(
+	ctx context.Context, sess *data.Session, intent *intents.IntentTyped[intents.IntentDataGetIdToken],
+) (*intents.IntentResponseIdToken, error) {
+	tnt := tenant.FromContext(ctx)
+
+	if len(intent.Data.Nonce) > 128 {
+		return nil, fmt.Errorf("invalid nonce")
+	}
+
+	sessData, _, err := crypto.DecryptData[*proto.SessionData](ctx, sess.EncryptedKey, sess.Ciphertext, tnt.KMSKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	var identity proto.Identity
+	if err := identity.FromString(sessData.Identity); err != nil {
+		return nil, fmt.Errorf("parsing session identity: %w", err)
+	}
+
+	acc, found, err := s.Accounts.Get(ctx, tnt.ProjectID, identity)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, fmt.Errorf("account not found")
+	}
+
+	walletAddr, err := AddressForUser(ctx, tnt, acc.UserID)
+	if err != nil {
+		return nil, fmt.Errorf("getting wallet address: %w", err)
+	}
+
+	aud := fmt.Sprintf("%s/project/%d", s.Config.Builder.BaseURL, tnt.ProjectID)
+	iat := time.Now()
+	exp := iat.Add(10 * time.Minute)
+
+	b := jwt.NewBuilder().
+		Subject(walletAddr).
+		Audience([]string{aud}).
+		Issuer(s.Config.BaseURL).
+		IssuedAt(iat).
+		Expiration(exp).
+		Claim("auth_time", sessData.CreatedAt.Unix()).
+		Claim(s.Config.BaseURL+"/identity", identity)
+
+	if acc.Email != "" {
+		b.Claim("email", acc.Email)
+	}
+
+	if intent.Data.Nonce != "" {
+		b.Claim("nonce", intent.Data.Nonce)
+	}
+
+	tok, err := b.Build()
```

Comment:
minor nit: `token` vs `tok` 

imho, saving few characters doesn't improve readability very much

I'd go for
- acc -> account
- t -> token
- h -> headers
- b -> jwt / jwtBuilder
- mh -> data / b / buf
- tnt -> tenant

Reply:
good suggestions, thanks! though in some cases (most notably `tnt`/`tenant`) it is required to not shadow the package name